recipes/tt-metalium/meta.yaml

```
@@ -0,0 +1,67 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+# If your package is python based, we recommend using Grayskull to generate it instead:
+# https://github.com/conda-incubator/grayskull
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 16 is convenient
+# when copying and pasting from another recipe, but not really needed.
+{% set name = "tt-metalium" %}
+{% set version = "0.55.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/tenstorrent/tt-metal/releases/download/v{{ version }}-rc13/tt-metalium.tar.gz
+    sha256: 5b4b4454c6fe60e91730b676ccc126fdc41149738e06f3e8d34bb9979bf30259
+
+build:
+  number: 0
+  skip: true  # [not linux or py!=310]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
+    - cmake
+    - ninja
+    - git
+
+  host:
+    - python
+    - pip
+    - wheel
+    - setuptools =69.5.1
```

Could this pin be relaxed?

```suggestion
    - setuptools
```

---

Initially it was not constrained, and I noticed absurd numbers of warnings. 

Probably some poor practices in our setup.py.

I pinned it to a tested version see if the warnings go away.

example of warning:


home/conda/staged-recipes/build_artifacts/tt-metalium_1737673554966/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.10/site-packages/setuptools/command/build_py.py:212: _Warning: Package 'tt_metal.common' is absent from the `packages` configuration.
2025-01-24T01:28:27.9339293Z       !!
2025-01-24T01:28:27.9339614Z       
2025-01-24T01:28:27.9516044Z               ********************************************************************************
2025-01-24T01:28:27.9520477Z               ############################
2025-01-24T01:28:27.9525248Z               # Package would be ignored #
2025-01-24T01:28:27.9528979Z               ############################
2025-01-24T01:28:27.9532709Z               Python recognizes 'tt_metal.common' as an importable package[^1],
2025-01-24T01:28:27.9537754Z               but it is absent from setuptools' `packages` configuration.
2025-01-24T01:28:27.9542179Z       
2025-01-24T01:28:27.9547546Z               This leads to an ambiguous overall configuration. If you want to distribute this
2025-01-24T01:28:27.9552078Z               package, please make sure that 'tt_metal.common' is explicitly added
2025-01-24T01:28:27.9557236Z               to the `packages` configuration field.
2025-01-24T01:28:27.9557656Z       
2025-01-24T01:28:27.9560939Z               Alternatively, you can also rely on setuptools' discovery methods
2025-01-24T01:28:27.9564825Z               (for example by using `find_namespace_packages(...)`/`find_namespace:`
2025-01-24T01:28:27.9569172Z               instead of `find_packages(...)`/`find:`).
2025-01-24T01:28:27.9573084Z       
2025-01-24T01:28:27.9577978Z               You can read more about "package discovery" on setuptools documentation page:
2025-01-24T01:28:27.9582034Z       
2025-01-24T01:28:27.9586458Z               - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html
2025-01-24T01:28:27.9590206Z       
2025-01-24T01:28:27.9595451Z               If you don't want 'tt_metal.common' to be distributed and are
2025-01-24T01:28:27.9600171Z               already explicitly excluding 'tt_metal.common' via
2025-01-24T01:28:27.9603896Z               `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
2025-01-24T01:28:27.9607331Z               you can try to use `exclude_package_data`, or `include-package-data=False` in
2025-01-24T01:28:27.9615269Z               combination with a more fine grained `package-data` configuration.
2025-01-24T01:28:27.9615703Z       
2025-01-24T01:28:27.9616050Z               You can read more about "package data files" on setuptools documentation page:
2025-01-24T01:28:27.9616552Z       
2025-01-24T01:28:27.9624712Z               - https://setuptools.pypa.io/en/latest/userguide/datafiles.html
2025-01-24T01:28:27.9625050Z       
2025-01-24T01:28:27.9625310Z       
2025-01-24T01:28:27.9625638Z               [^1]: For Python, any directory (with suitable naming) can be imported,
2025-01-24T01:28:27.9636130Z                     even if it does not contain any `.py` files.
2025-01-24T01:28:27.9637605Z                     On the other hand, currently there is no concept of package data
2025-01-24T01:28:27.9637978Z                     directory, all directories are treated like packages.
2025-01-24T01:28:27.9638373Z               ********************************************************************************
2025-01-24T01:28:27.9643265Z       
2025-01-24T01:28:27.9643549Z       !!
2025-01-24T01:28:27.9643835Z         check.warn(importable)
2025-01-24T01:28:27.9644598Z       /home/conda/staged-recipes/build_artifacts/tt-metalium_1737673554966/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.10/site-packages/setuptools/command/build_py.py:212: _Warning: Package 'tt_metal.hostdevcommon.api.hostdevcommon' is absent from the `packages` configuration.
2025-01-24T01:28:27.9648696Z       !!
2025-01-24T01:28:27.9648997Z       
2025-01-24T01:28:27.9649344Z               ********************************************************************************
2025-01-24T01:28:27.9649696Z               ############################
2025-01-24T01:28:27.9649924Z               # Package would be ignored #
2025-01-24T01:28:27.9652893Z               ############################
2025-01-24T01:28:27.9653279Z               Python recognizes 'tt_metal.hostdevcommon.api.hostdevcommon' as an importable package[^1],
2025-01-24T01:28:27.9653694Z               but it is absent from setuptools' `packages` configuration.
2025-01-24T01:28:27.9654008Z       
2025-01-24T01:28:27.9658543Z               This leads to an ambiguous overall configuration. If you want to distribute this
2025-01-24T01:28:27.9658987Z               package, please make sure that 'tt_metal.hostdevcommon.api.hostdevcommon' is explicitly added
2025-01-24T01:28:27.9659378Z               to the `packages` configuration field.
2025-01-24T01:28:27.9659930Z       
2025-01-24T01:28:27.9663458Z               Alternatively, you can also rely on set

---

the warning is no a good enough reason to take such an old version; though of course it would be good to react to those warnings in the upstream project

---

Ok, I'll aim to resolve the warnings a different way.